soc/silabs/silabs_siwx91x/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2024 Silicon Laboratories Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+add_subdirectory(siwg917)
+
+# Necessary to not overwrite NWP Firmware
+math(EXPR FLASH_LOAD_ADDRESS "(${CONFIG_FLASH_BASE_ADDRESS}) + (${CONFIG_FLASH_LOAD_OFFSET})")
+
+set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+    COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/siwx91x_isp_prepare.py
+    --load-addr ${FLASH_LOAD_ADDRESS}
+    ${KERNEL_BIN_NAME}
+    ${KERNEL_BIN_NAME}.rps
+)
```

Comment:
don't see how this is going to work if someone just enables e.g. mcuboot or if someone uses sysbuild? Why is this needed? Why can't DTS be setup correctly to account for this?

Reply:
> Why is this needed?

The long story short, flashing the siwx91x requires a specific file format.

> how this is going to work if someone just enables e.g. mcuboot or if someone uses sysbuild? 

MCUboot is indeed incompatible.

> Why can't DTS be setup correctly to account for this?

This requirement is not about the memory layout, it is about the file format.

The technical details:

The main CPU (or the JTag) is not able to write the flash. The write is made by the auxiliary CPU (aka "NWP"). NWP has it own bootloader in a ROM. So, flashing the siwx91x means sending the firmware payload (using the .rps format) over the UART to the NWP bootloader.

(I could add this explanation in soc/silabs/silabs_siwx91x/siwx91x_isp_prepare.py)